src/commonTest/kotlin/com/configcat/ConfigCatClientTests.kt

```diff
@@ -510,13 +529,260 @@ class ConfigCatClientTests {
             pollingMode = manualPoll()
         }
 
-        client.refresh()
+        client.forceRefresh()
         assertEquals(
             "ConfigCat-Kotlin/m-${Constants.version}",
             mockEngine.requestHistory.last().headers["X-ConfigCat-UserAgent"]
         )
     }
 
+    @Test
+    fun testGetValueDetailsWithError() = runTest {
+        val mockEngine = MockEngine {
+            respond(
+                content = "",
+                status = HttpStatusCode.BadRequest
+            )
+        }
+        val client = ConfigCatClient("test") {
+            httpEngine = mockEngine
+        }
+
+        val details = client.getValueDetails("fakeKey", "")
+
+        assertEquals("", details.value)
+        assertTrue(details.isDefaultValue)
+        assertEquals("Config JSON is not present. Returning defaultValue: ''.", details.error)
+    }
+
+    @Test
+    fun testOnlineOffline() = runTest {
+        val mockEngine = MockEngine {
+            respond(
+                content = Data.formatJsonBody(true),
+                status = HttpStatusCode.OK
+            )
+        }
+        val client = ConfigCatClient("test") {
+            httpEngine = mockEngine
+            pollingMode = autoPoll { pollingInterval = 2.seconds }
+        }
+
+        TestUtils.awaitUntil {
+            mockEngine.requestHistory.size == 1
+        }
+
+        client.setOffline()
+        assertTrue(client.isOffline)
+
+        client.forceRefresh()
+
+        assertEquals(1, mockEngine.requestHistory.size)
+
+        client.setOnline()
+        assertFalse(client.isOffline)
+
+        TestUtils.awaitUntil {
+            mockEngine.requestHistory.size > 1
+        }
+    }
+
+    @Test
+    fun testInitOffline() = runTest {
+        val mockEngine = MockEngine {
+            respond(
+                content = Data.formatJsonBody(true),
+                status = HttpStatusCode.OK
+            )
+        }
+        val client = ConfigCatClient("test") {
+            httpEngine = mockEngine
+            pollingMode = autoPoll { pollingInterval = 2.seconds }
+            offline = true
+        }
+
+        assertTrue(client.isOffline)
+
+        client.forceRefresh()
+
+        assertEquals(0, mockEngine.requestHistory.size)
+
+        client.setOnline()
+        assertFalse(client.isOffline)
+
+        TestUtils.awaitUntil {
+            mockEngine.requestHistory.size > 1
+        }
+    }
+
+    @Test
+    fun testDefaultUser() = runTest {
```

Comment:
I think we should test getVarationId too.

Reply:
Done